### PR TITLE
Preservation of data

### PR DESCRIPTION
**Pull request by Kakalanp**
Adding new methods to different files:
- book_class:
  - add_json_book

- student_class:
  - add_json_student

- teacher_class:
  - add_json_teacher

- rental_class:
  - add_json_rental

- app:
  - read_book_json
  - read_people_json
  - read_rentals_json
  - save_book
  - save_person
  - save_rental

### CODE REVIEW

**Review by Meltrust — CHANGES_REQUESTED** [review]
Hi @Kakalanp,

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!



### To highlight:

- The app works✔️ 

You can do it

## Required Changes ♻️

- [x] On your readme file, please, kindly add the remaining sections that you are missing, according to the [Microverse Readme Template](https://github.com/microverseinc/readme-template).  For almost all projects in Microverse, all the sections are required, except for "tests" and "live demo" if not required by the project.

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Comment by richardoppiyo** [comment]
I have updated the README file in the above commit

File: README.md [hunk 1]
```diff
@@ -1,7 +1,31 @@
...
+ ## Built With
+
+ - Ruby
+
+ ## Getting Started
+
+ To get a local copy up and running follow these simple example steps.
+
+ `https://github.com/Kakalanp/OOP-school-library.git`
+
```
Comment (Gambit142) [inline README.md]: - [ ] For the getting started, we just have a URL, there are no instructions for cloning the budget and navigating to the root directory of the project. For example:
```
git clone <URL>
cd <name of repository here>
open with a code editor
```

#### Kindly implement this 👍

**Review by Gambit142 — CHANGES_REQUESTED** [review]
Hi @Kakalanp @richardoppiyo 👋, 

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

 ### Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 1 above)
Comment (richardoppiyo) [reply README.md]: Noted.
The changes you mentioned have been implemented.

File: README.md [hunk 2]
```diff
@@ -1,7 +1,34 @@
...
+ - Ruby
+
+ ## Getting Started
+
+ To get a local copy up and running follow these simple example steps.
+- git clone  `https://github.com/Kakalanp/OOP-school-library.git`
+
+- change to the directory using `cd OOP-school-library`
+
+- open with your prefffered code editor.
```
Comment (Mihndim2020) [inline README.md]: - [OPTIONAL] Please there is a typing error with `preferred`, make sure you correct that.

**Review by Mihndim2020 — APPROVED** [review]
# STATUS: APPROVED!!!  🎉 🎉 🎉 :champagne: :champagne: :champagne:
Hi Team, 

Your project is complete! Great work on making the changes requested by the previous reviewer. There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 🎉 🎉 :champagne: :champagne: :champagne:

Congratulations-Memes-FEATURED

## Optional suggestions

_Check the comments under the review._

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: README.md (hunk 2 above)
Comment (richardoppiyo) [reply README.md]: Noted, and thank you!

**Pull request merged**